src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -131,21 +129,29 @@ public IEnumerable<ITagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotSp
                     var dataTagSpans = tag.Span.GetSpans(snapshot);
                     if (dataTagSpans.Count == 1)
                     {
-                        var dataTagSpan = dataTagSpans[0];
-                        var parameterHintUITag = InlineHintsTag.Create(
-                            tag.Tag.Hint, Format, _textView, dataTagSpan, _taggerProvider, _formatMap, classify);
-
-                        _cache.Add(new TagSpan<IntraTextAdornmentTag>(dataTagSpan, parameterHintUITag));
+                        _cache.Add((tag, null));
                     }
                 }
             }
 
+            var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
+            var classify = document?.Project.Solution.Workspace.Options.GetOption(InlineHintsOptions.ColorHints, document?.Project.Language) ?? false;
+
             var selectedSpans = new List<ITagSpan<IntraTextAdornmentTag>>();
-            foreach (var tagSpan in _cache)
+            for (var i = 0; i < _cache.Count; i++)
```

Comment:
```suggestion
            foreach (var (mappingTagSpan, tagSpan) in _cache)
```

Reply:
The index is required for assigning a value back.